mite_selenium/__init__.py

```diff
@@ -43,19 +47,41 @@ def _spec_import_if_none(self, config_option):
 
     def _start(self):
         self._context.browser = self
-        self._remote = Remote(
-            desired_capabilities=self._capabilities,
-            command_executor=self._command_executor,
-            browser_profile=self._browser_profile,
-            proxy=self._proxy,
-            keep_alive=self._keep_alive,
-            file_detector=self._file_detector,
-            options=self._options,
-        )
+        try:
+            if self._is_selenium_wire:
```

Comment:
Is there a usecase for not having selenium wire?  If not, we should just use it unconditionally.  The thing I can most reasily imagine is if selenium-wire is much slower than selenium itself (medium-bad) and if it's not possible to measure and subtract out that extra slowness from our metrics (very bad indeed).

Reply:
I don't know much about selenium wire, but I do have the general principle that we should not assume what the end user of mite is going to want. I think if there were any side effect, it would make this decision for us.

Reply:
In the selenium-wire docs it states 
> Selenium Wire has limited support for using the remote webdriver client.
So a user might need some functionality that is not available yet. Also like Jordan said about not assuming what the end user needs from mite.
I also think it's too early for us to tell if selenium-wire is much slower that selenium itself and if it's possible or not to measure and subtract that out of our metrics and so I would keep both for now.

Reply:
Yeah, I agree with what Jordan is saying about avoiding side effects.  And the docs Cosmin quoted show that there are potential side effects (missing functionality).  So we need to support both.  In that case, I think:
1. the selenium wire specific code should be in a subclass of `_SeleniumWrapper` rather than in conditionals
2. the `mite_selenium` decorator should gain an `wire=True/False` argument that controls whether it returns a `_SeleniumWrapper` or a `_SeleniumWireWrapper`
3. (not strictly necessary but nice to have) we should make selenium wire an [optional dependency](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#optional-dependencies) so that it's not pulled in unless the user specifically requests it (obviously, they can't use the wire functionality unless the dependency is present -- we should detect that case in the `mite_selenium` decorator and raise an error)